backend/x86_gas.ml

```diff
@@ -280,7 +280,23 @@ let print_line b = function
   | Byte n -> bprintf b "\t.byte\t%a" cst n
   | Bytes s ->
       if system = S_solaris then buf_bytes_directive b ".byte" s
-      else bprintf b "\t.ascii\t\"%s\"" (string_of_string_literal s)
+      else
+        (* Very long lines can cause gas to be extremely slow so split up large
+          string literals. It turns out that gas reads files in 32kb chunks
+          so splitting the string into blocks of 25k characters should be close
+          to the sweet spot even with a lot of escapes. *)
+        let chunk_size = 25000 in
```

Comment:
(I wonder whether we should play it safe
and consider that each character will be
output using the `\012` form.)

Reply:
It doesn't seem to get noticeably slower until close to 1MB chunks so I don't think it really matters.